samples/workload/tensorflow/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2019 Intel Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+cmake_minimum_required (VERSION 3.14)
+
+project(tflite_wasm)
+
+include(ExternalProject)
+
+# Only Download
+ExternalProject_Add(tensorflow-download
+    PREFIX tensorflow
+    GIT_REPOSITORY https://github.com/tensorflow/tensorflow.git
+    GIT_TAG        v2.13.0-rc2
+    GIT_PROGRESS   ON
+    SOURCE_DIR     ${CMAKE_CURRENT_SOURCE_DIR}/tensorflow
+    UPDATE_COMMAND git restore .
```

Comment:
The `UPDATE_COMMAND` is invoked after the `PATCH_COMMAND`, which immediately rolls back the changes of the patch and leads to the issue encountered in https://github.com/bytecodealliance/wasm-micro-runtime/issues/2351#issuecomment-1835772846.
A suggestion (which worked for me) was moving `git restore .` as the first command run by the patch command, and remove the update command.

Reply:
May I ask:
- what's the root cause of #2351
- What's your local cmake version

`UPDATE_COMMAND` may have different behavior in variant cmake building. It will be confused.